core/cmdbchangeop.class.inc.php

```diff
@@ -63,22 +63,71 @@ public static function Init()
 
 	/**
 	 * Describe (as a text string) the modifications corresponding to this change
-	 */	 
+	 */
 	public function GetDescription()
 	{
 		return '';
 	}
 
 	/**
-	 * Safety net: in case the change is not given, let's guarantee that it will
-	 * be set to the current ongoing change (or create a new one)	
-	 */	
+	 * @param string $sAttCode
+	 * @param mixed $value
+	 *
+	 * @return bool
+	 * @throws \CoreException
+	 * @throws \CoreUnexpectedValue
+	 *
+	 * @since 2.7.4 reject non persisted CMDBChange
+	 */
+	public function Set($sAttCode, $value)
+	{
+		if ($sAttCode !== 'change') {
+			return parent::Set($sAttCode, $value);
+		}
+
+		if (is_int($value)) {
+			if ($value <= 0) {
+				$this->ThrowInvalidCMDBChangeException();
+			}
+
+			return parent::Set($sAttCode, $value);
+		}
+
+		if (is_object($value)) {
+			if ($value->IsNew()) {
+				$this->ThrowInvalidCMDBChangeException();
+			}
+
+			return parent::Set($sAttCode, $value);
+		}
+
+		// we should never get here O:)
+		throw new CoreUnexpectedValue("Invalid value set to CMDBChangeOp.change");
```

Comment:
Again, now you made it worse. There are now 3 places where you call the exact same return.

To improve readability, I would suggest to revert the logic to:
```suggestion
		if ($sAttCode == 'change') {
			if (is_int($value) && $value <= 0) {
				$this->ThrowInvalidCMDBChangeException();
			}
			elseif (is_object($value) && $value->IsNew()) {
				$this->ThrowInvalidCMDBChangeException();
			}
		}

		return parent::Set($sAttCode, $value);
```

Reply:
I like your first proposal : exit condition first. Will change this !

Reply:
Sorry I misread on friday : main exit condition is already on top. After reading again and more carefully your proposition, I agree grouping the error conditions is better. 
I've just pushed 38acbc87

Reply:
Please see my suggestion again, the call to parent should not be inside an `if` for readability of the code..

Reply:
After a review with Romain (see last comment below), I will drop this to lower this PR's scope.
Thanks for your feedback !
I'm closing this conversation.